### PR TITLE
[canonical-livepatch-onprem] Add Canonical Livepatch on-prem server support

### PR DESCRIPTION
**Pull request by adam-vest**
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/canonical_livepatch_onprem.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    short_desc = 'Canonical Livepatch Onprem Service'
+
+    plugin_name = 'canonical_livepatch_onprem'
+    profiles = ('services',)
+    files = ('/usr/local/bin/livepatchd',)
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/livepatchd.yaml",
```
Comment (jcastill) [inline sos/report/plugins/canonical_livepatch_onprem.py:22]: lgtm, just a question: is there an api token or a user/pass we should obfuscate in this file?

**Comment by arif-ali** [comment]
if you added `services = ("livepatch-server")` for the trigger at the top instead of `files`, then both the service status and the journal would be added by default, and then you will not need the 2 status checks and `files`

File: sos/report/plugins/canonical_livepatch_onprem.py (hunk 1 above)
Comment (adam-vest) [reply sos/report/plugins/canonical_livepatch_onprem.py:22]: There are both, actually - user credentials and a token that authenticates against upstream livepatch.. User password is already encoded, but perhaps not a bad idea to obfuscate. Is that something we'd do in the plugin directly, or elsewhere in sos?

Comment (jcastill) [reply sos/report/plugins/canonical_livepatch_onprem.py:22]: You can do it directly in the plugin itself. Here's an example of obfuscation in a yaml file:

https://github.com/sosreport/sos/blob/main/sos/report/plugins/juju.py#L65

**Comment by adam-vest** [comment]
> if you added `services = ("livepatch-server")` for the trigger at the top instead of `files`, then both the service status and the journal would be added by default, and then you will not need the 2 status checks and `files`

Hey @arif-ali , I tried doing the following as (I think you) suggested:

```
    services = ("livepatch-server")

    def setup(self):
        self.add_copy_spec([
            "/etc/livepatchd.yaml",
        ])
```

...but I didn't see the journal or service status in the resultant sos report? Normally these were appearing in `sos_commands/canonical_livepatch_server`. Is this new as of 4.6.0? I'm just testing against the version in apt, which is `4.5.6`.

File: sos/report/plugins/canonical_livepatch_onprem.py (hunk 1 above)
Comment (adam-vest) [reply sos/report/plugins/canonical_livepatch_onprem.py:22]: Awesome, thanks for the link! I have some obfuscation code in now that looks like it's having the intended effect :slightly_smiling_face:

**Comment by TurboTurtle** [comment]
> > if you added `services = ("livepatch-server")` for the trigger at the top instead of `files`, then both the service status and the journal would be added by default, and then you will not need the 2 status checks and `files`
> 
> Hey @arif-ali , I tried doing the following as (I think you) suggested:
> 
> ```
>     services = ("livepatch-server")
> 
>     def setup(self):
>         self.add_copy_spec([
>             "/etc/livepatchd.yaml",
>         ])
> ```
> 
> ...but I didn't see the journal or service status in the resultant sos report? Normally these were appearing in `sos_commands/canonical_livepatch_server`. Is this new as of 4.6.0? I'm just testing against the version in apt, which is `4.5.6`.

You're missing a comma in `services` to make it a tuple, rather than a string. `services = ('livepatch-server',)` The implicit collection for `services` was definitely in 4.5, but I forget exactly when we added it.

**Comment by adam-vest** [comment]
> > > if you added `services = ("livepatch-server")` for the trigger at the top instead of `files`, then both the service status and the journal would be added by default, and then you will not need the 2 status checks and `files`
> > 
> > 
> > Hey @arif-ali , I tried doing the following as (I think you) suggested:
> > ```
> >     services = ("livepatch-server")
> > 
> >     def setup(self):
> >         self.add_copy_spec([
> >             "/etc/livepatchd.yaml",
> >         ])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ...but I didn't see the journal or service status in the resultant sos report? Normally these were appearing in `sos_commands/canonical_livepatch_server`. Is this new as of 4.6.0? I'm just testing against the version in apt, which is `4.5.6`.
> 
> You're missing a comma in `services` to make it a tuple, rather than a string. `services = ('livepatch-server',)` The implicit collection for `services` was definitely in 4.5, but I forget exactly when we added it.

Awesome, got it! Thanks for the feedback - pushed an update to implement these things. Looking forward to any more feedback!

**Review by TurboTurtle — APPROVED** [review]
LGTM

**Review by arif-ali — APPROVED** [review]

**Pull request merged**